File: Python3/프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    # 1. 전화번호부를 정렬
    phone_book.sort()
    
    # 2. 인접한 두 전화번호를 비교하여 접두어 관계가 있는지 확인
    for i in range(len(phone_book) - 1):
        if phone_book[i + 1].startswith(phone_book[i]):
            return False  # 접두어 관계가 있으면 False 반환
    
    # 접두어 관계가 없다면 True 반환
    return True

File: C++/프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    sort(phone_book.begin(), phone_book.end());
    for (int i = 1; i<phone_book.size(); i++){
            if (phone_book[i].find(phone_book[i-1]) ==0){
                answer = false;
                return answer;
            }
    }
    //pyton의 startswith > find() ==0 endswith >> rfind() == str.length() - tar.length() 
    return answer;
}